be/src/util/block_compression.cpp

```diff
@@ -116,13 +121,11 @@ class Lz4BlockCompression : public BlockCompressionCodec {
                     input.size);
         }
 
-        Context* context;
-        RETURN_IF_ERROR(_acquire_compression_ctx(&context));
+        std::unique_ptr<Context> context;
+        RETURN_IF_ERROR(_acquire_compression_ctx(context));
         bool compress_failed = false;
         Defer defer {[&] {
-            if (compress_failed) {
-                _delete_compression_ctx(context);
-            } else {
+            if (!compress_failed) {
                 _release_compression_ctx(context);
```

Comment:
use std::move(context). Then the developer will know the context ptr is invalid now.